pkg/controllers/walruscore/connector.go

```diff
@@ -3,29 +3,142 @@ package walruscore
 import (
 	"context"
 
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/workqueue"
 	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlcli "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlevent "sigs.k8s.io/controller-runtime/pkg/event"
+	ctrlhandler "sigs.k8s.io/controller-runtime/pkg/handler"
 	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
 	ctrlreconcile "sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	walruscore "github.com/seal-io/walrus/pkg/apis/walruscore/v1"
+	"github.com/seal-io/walrus/pkg/apistatus"
 	"github.com/seal-io/walrus/pkg/controller"
+	"github.com/seal-io/walrus/pkg/resourcehandler"
+	"github.com/seal-io/walrus/pkg/resourcehandlers"
+	"github.com/seal-io/walrus/pkg/systemmeta"
 )
 
 // ConnectorReconciler reconciles a v1.Connector object.
-type ConnectorReconciler struct{}
+type ConnectorReconciler struct {
+	client ctrlcli.Client
+}
 
 var _ ctrlreconcile.Reconciler = (*ConnectorReconciler)(nil)
 
 func (r *ConnectorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = ctrllog.FromContext(ctx)
+	logger := ctrllog.FromContext(ctx)
+
+	obj := new(walruscore.Connector)
+	err := r.client.Get(ctx, req.NamespacedName, obj)
+	if err != nil {
+		return ctrl.Result{}, ctrlcli.IgnoreNotFound(err)
+	}
+
+	// Skip.
+	{
+		// Skip deletion.
+		if obj.DeletionTimestamp != nil {
+			return ctrl.Result{}, nil
+		}
+
+		// Skip connector already reconciling.
+		if apistatus.ConnectorConditionConnected.IsUnknown(obj) && !apistatus.ConnectorConditionRefresh.IsUnknown(obj) {
+			return ctrl.Result{}, nil
+		}
+
+		// Skip connector finished reconciling.
+		if (apistatus.ConnectorConditionConnected.IsTrue(obj) || apistatus.ConnectorConditionConnected.IsFalse(obj)) &&
+			!apistatus.ConnectorConditionRefresh.IsUnknown(obj) {
+			return ctrl.Result{}, nil
+		}
+	}
+
+	// Initialize status.
+	{
+		apistatus.ConnectorConditionConnected.Unknown(obj, apistatus.ConnectorConditionReasonConnecting, "Connecting")
+		if apistatus.ConnectorConditionRefresh.IsUnknown(obj) {
+			apistatus.ConnectorConditionRefresh.False(obj, "", "")
+		}
+		err = r.client.Status().Update(ctx, obj)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
```

Comment:
we should give a judgment to evaluate when to initialize the status's conditions, however, we cannot do that in reconciling only. we must leverage admission webhooks.

1. add webhook-gen marker with `subResources=["status"]`
2. validating webhook `ValidateUpdate` to check
  - if the old conditions are completed, like ConnectorConditionConnected == True/False, then allows to change to Unknown. 
  - if the old conditions are processing, like ConnectorConditionConnected == Unknown, then allows to change to True/False.
3. Predicate with `if the generation is changed OR some conditions are Unknown`
  - increasing generation means Spec changing.
  - `ConnectorConditionConnected == Unknown` or the conditions are nil/empty means the User make changing, like we `kubectl edit xxx --subresource=status` to modify the conditions.
4. Skip only if the object is deleted without locking.
  - if the object is deleted and still locked, we should remove the related secret.

Now the reconciling object must be Spec changing or Status initializing.
1. if it's Spec changing, we can do conditions initialization, like give a ConnectorConditionConnected: Unknown to the condition slice, and do `client.Status().Update()`.
2. Keep going with connectivity verification, do `client.Status().Update()` to update the completed status.